src/main/java/net/openhft/hashing/LongTupleHashFunction.java

```diff
@@ -0,0 +1,779 @@
+package net.openhft.hashing;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import javax.annotation.ParametersAreNonnullByDefault;
+import sun.nio.ch.DirectBuffer;
+
+import java.io.Serializable;
+import java.nio.ByteBuffer;
+
+import static net.openhft.hashing.CharSequenceAccess.nativeCharSequenceAccess;
+import static net.openhft.hashing.UnsafeAccess.*;
+import static net.openhft.hashing.Util.*;
+
+/**
+ * Tuple hash function producing {@code long[]} result from byte sequences of any length and
```

Comment:
This class-level comment largely copies that of `LongHashFunction`. It will be hard to support both. Better just to say something like "see the definition of byte sequence semantics in the javadoc for `{@link LongHashFunction`", and then focus this comment on what makes `LongTupleHashFunction` different from `LongHashFunction`.